src/gui/README.md

```diff
@@ -0,0 +1,285 @@
+# Graphical User Interface
+
+The graphical user interface can be access by launching OpenROAD with ``-gui``.
+
+## Commands
+
+### Add buttons to the toolbar
+
+```
+create_toolbar_button [-name name]
+                      -text button_text
+                      -script tcl_script 
+                      [-echo]
+```
+
+Returns: name of the new button, either ``name`` or ``buttonX``.
+
+Options description:
+- ``button_text``: The text to put on the button.
+- ``tcl_script``: The tcl script to evaluate when the button is pressed.
+- ``name``: (optional) name of the button, used when deleting the button.
+- ``echo``: (optional) indicate that the commands in the ``tcl_script`` should be echoed in the log.
+
+To remove the button: 
+
+```
+gui::remove_toolbar_button name
+```
+
+## TCL functions
+
+### Support
+
+Determine is the GUI is active:
+
+```
+gui::enabled
+```
+
+Announce to the GUI that a design was loaded:
+
+```
+gui::design_created
+```
+
+To load the results of a DRC report:
+
+```
+gui::load_drc filename
+```
+
+### Layout navigation
+
+To fit the whole layout in the window:
+
+```
+gui::fit
+```
+
+To zoom in our out to a specific region:
+
+```
+gui::zoom_to x0 y0 x1 y1
+```
+
+Options description:
+- ``x0, y0`` first corner of the layout area in microns.
+- ``x1, y1`` second corner of the layout area in microns.
+
+To zoom in the layout:
+
+```
+gui::zoom_in
+gui::zoom_in x y
+```
+
+Options description:
+- ``x, y`` new center of layout
+
+To zoom out the layout:
+
+```
+gui::zoom_out
+gui::zoom_out x y
+```
+
+Options description:
+- ``x, y`` new center of layout
+
+To move the layout to new area:
+
+```
+gui::center_at x y
+```
+
+Options description:
+- ``x, y`` new center of layout
+
+To change the resolution to a specific value:
+
+```
+gui::set_resolution resolution
+```
+
+Options description:
+- ``resolution`` database units per pixel
+
+### Selections
+
+To add a single net to the selected items:
+
+```
+gui::selection_add_net name
+```
+
+Options description:
+- ``name`` name of the net to add
+
+To add several nets to the selected items:
+
+```
+gui::selection_add_nets name
```

Comment:
name_regex

Reply:
I notice that this doesn't actually work as 

```
  void addSelectedNets(const char* pattern,
...
                       bool match_reg_ex = false,
```

though
```
  void addSelectedInsts(const char* pattern,
...
                        bool match_regE_ex = true,
```

Perhaps you could fix this along the way.

Reply:
no problem